### PR TITLE
Change structure project for "cazanw" binary at installation of cazanw with "npm install"

### PR DESCRIPTION
**Pull request by Sellig6792**

### CODE REVIEW

File: post-build/src/cazanw.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+if (platform === 'win32') {
+return arch === 'arm64' ? 'aarch64-pc-windows-msvc' : 'x86_64-pc-windows-gnu';
+}
+
+if (platform === 'linux') {
+return arch === 'arm64' ? 'aarch64-unknown-linux-gnu' : 'x86_64-unknown-linux-gnu';
+}
+
+throw new Error(`Unsupported platform or architecture: ${platform} ${arch}`);
+}
```
Comment (hamirmahal) [inline post-build/src/cazanw.js:27]:
```suggestion
function getWhichTarget() {
  if (platform === "darwin") {
    return arch === "arm64" ? "aarch64-apple-darwin" : "x86_64-apple-darwin";
  }

  if (platform === "win32") {
    return arch === "arm64"
      ? "aarch64-pc-windows-msvc"
      : "x86_64-pc-windows-gnu";
  }

  if (platform === "linux") {
    return arch === "arm64"
      ? "aarch64-unknown-linux-gnu"
      : "x86_64-unknown-linux-gnu";
  }

  throw new Error(`Unsupported platform or architecture: ${platform} ${arch}`);
}
```

**Review by hamirmahal — CHANGES_REQUESTED** [review]

**Comment by Sellig6792** [comment]
Anything else @hamirmahal 😄😄?

File: post-build/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+use std::fs;
+use std::path::PathBuf;
+
+fn open_package_json() -> serde_json::Map<String, serde_json::Value> {
```
Comment (hamirmahal) [inline post-build/src/main.rs:4]: Hmm... I don't see an import for `serde_json`. Is that intentional?

Comment (Sellig6792) [reply post-build/src/main.rs:4]: In this file: `post-build/Cargo.toml`

**Comment by Sellig6792** [comment]
I'll implement the prebuild before merging the PR

**Pull request closed without merging**